android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -111,71 +117,80 @@ constructor(
 
     var carePlanModified = false
 
-    planDefinition.action.forEach { action ->
-      val input = Bundle().apply { entry.addAll(data.entry) }
-
-      if (action.passesConditions(input, planDefinition, subject)) {
-        val definition = action.activityDefinition(planDefinition)
-
-        if (action.hasTransform()) {
-          val taskPeriods = action.taskPeriods(definition, output)
-
-          taskPeriods.forEachIndexed { index, period ->
-            val source =
-              Parameters().apply {
-                addResourceParameter(CarePlan.SP_SUBJECT, subject)
-                addResourceParameter(PlanDefinition.SP_DEFINITION, definition)
-                // TODO find some other way (activity definition based) to pass additional data
-                addResourceParameter(PlanDefinition.SP_DEPENDS_ON, data)
-              }
-            source.setParameter(Task.SP_PERIOD, period)
-            source.setParameter(ActivityDefinition.SP_VERSION, IntegerType(index))
-
-            val structureMap = fhirEngine.get<StructureMap>(IdType(action.transform).idPart)
-            structureMapUtilities.transform(
-              transformSupportServices.simpleWorkerContext,
-              source,
-              structureMap,
-              output,
-            )
+    if (generateCarePlanWithWorkflowApi) {
+      workflowCarePlanGenerator.applyPlanDefinitionOnPatient(
+        planDefinition = planDefinition,
+        patient = subject as Patient,
+        output = output,
+      )
+    } else {
+      planDefinition.action.forEach { action ->
+        val input = Bundle().apply { entry.addAll(data.entry) }
+
+        if (action.passesConditions(input, planDefinition, subject)) {
+          val definition = action.activityDefinition(planDefinition)
+
+          if (action.hasTransform()) {
+            val taskPeriods = action.taskPeriods(definition, output)
+
+            taskPeriods.forEachIndexed { index, period ->
+              val source =
+                Parameters().apply {
+                  addResourceParameter(CarePlan.SP_SUBJECT, subject)
+                  addResourceParameter(PlanDefinition.SP_DEFINITION, definition)
+                  // TODO find some other way (activity definition based) to pass additional data
+                  addResourceParameter(PlanDefinition.SP_DEPENDS_ON, data)
+                }
+              source.setParameter(Task.SP_PERIOD, period)
+              source.setParameter(ActivityDefinition.SP_VERSION, IntegerType(index))
+
+              val structureMap = fhirEngine.get<StructureMap>(IdType(action.transform).idPart)
+              structureMapUtilities.transform(
+                transformSupportServices.simpleWorkerContext,
+                source,
+                structureMap,
+                output,
+              )
+            }
```

Comment:
can we put this entire branch path in a `generateOrUpdateCarePlanFromStructureMap` function?

whoops I guess not just these lines but all the way down, this entire `else` branch

Reply:
done